### ISSUE TITLE
Impossible to declare enum case (String) raw value with the same name as the case, without warning

### ISSUE THREAD

**Issue by dejanskledar** [realm/SwiftLint#2111]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

For **Trigger no.1**:
```bash
$ swiftlint lint
warning: Explicit Enum Raw Value Violation: Enums should be explicitly assigned their raw values. (explicit_enum_raw_value)
```

For **Trigger no.2**:
```bash
$ swiftlint lint
warning: Redundant String Enum Value Violation: String enum values can be omitted when they are equal to the enumcase name. (redundant_string_enum_value)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.25.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
CocoaPods

* Which Xcode version are you using (check `xcode-select -p`)?
Xcode 9.2 (9C40b)

```swift
// This triggers no.1 (Trigger no.1):
enum Something: String {
    case foo
}

// This triggers no.2 (Trigger no.2):
enum Something: String {
    case foo = "foo"
}
```

As it seems, you can't write an String enum case with the same string value as the case name.

Probably should add an exception to the `explicit_enum_raw_value` for String enum values.

**Comment by marcelofabri** [comment]
Those rules shouldn't be enabled at the same time - they basically enforce opposite patterns. That's why `explicit_enum_raw_value` is opt-in.

**Comment by marcelofabri** [comment]
Closing this for now. Feel free to reopen if you have any questions.

**Comment by lordzsolt** [comment]
I ran into this issue just now.

I was thinking of adding a configuration option, to exclude enums with `String`  rawValue from `explicit_enum_raw_value`.
@marcelofabri thoughts?
